src/transformers/trainer.py

```diff
@@ -5094,6 +5096,14 @@ def create_accelerator_and_postprocess(self):
             args["dataloader_config"] = dataloader_config
         else:
             args.update(accelerator_config)
+        # tp is initialized at Accelerator init phase so
+        # args should be prepared here
+        if self.args.tp_size > 1:
+            self.is_tp_enabled = True
+            if version.parse(accelerate_version) > version.parse("1.3.0"):
+                args["torch_tp_plugin"] = TorchTensorParallelPlugin(tp_size=self.args.tp_size)
```

Comment:
I am not familiar with this API, so we need some documentation about what this uses under the hood! 
Also we could check if model supports TP? or is it not even needed?

Reply:
> Also we could check if model supports TP? or is it not even needed?


Hi @ArthurZucker We check this in accelerate here - https://github.com/huggingface/accelerate/blob/526925b48c07d997cdd9bf5911f659ca45778915/src/accelerate/accelerator.py#L1511

Reply:
@ArthurZucker 

> I am not familiar with this API, so we need some documentation about what this uses under the hood!



Do you want me to add that as a comment above this piece of code or please point me to a place in HF docs where you want me to add it. Thank you.

Reply:
but interesting did not know accelerate already supported it!

Reply:
@ArthurZucker We added the support to accelerate at this PR - https://github.com/huggingface/accelerate/pull/3173